### PR TITLE
Fixing Libiio build

### LINKED ISSUE #124948: libiio fails to build
**Describe the bug**
`libiio` fails to build on master/release-21.05. The build failure (`libxml2` is not found anymore by cmake) appeared with the update of `libxml2`  from 2.9.10 to 2.9.12.

Addressing https://github.com/NixOS/nixpkgs/issues/122042

CC @thoughtpolice

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Closes https://github.com/NixOS/nixpkgs/issues/124948

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Not sure why but this fixed the build.
It seems similar to https://github.com/bincrafters/community/issues/930.

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124951` at e26ad6c6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>libiio</li>
    <li>sdrangel</li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.libiio</li>
    <li>python39Packages.libiio</li>
    <li>urh</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/development/libraries/libiio/default.nix:26:15:
```
   |
26 |   patches = [ ./cmake-fix-libxml2-find-package.patch ];
   |               ^
```
</li>
  </ul>
</details>

**Comment by markuskowa** [comment]
Thanks for the fix! Can you please base it against `staging`?

File: pkgs/development/libraries/libiio/cmake-fix-libxml2-find-package.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --color -ur a/CMakeLists.txt b/CMakeLists.txt
+--- a/CMakeLists.txt	2021-05-30 13:46:22.256040282 +0200
++++ b/CMakeLists.txt	2021-05-30 14:15:42.530181216 +0200
+@@ -333,7 +333,7 @@
+ # So, try first to find the CMake module provided by libxml2 package, then fallback
+ # on the CMake's FindLibXml2.cmake module (which can lack some definition, especially
+ # in static build case).
+-find_package(LibXml2 QUIET NO_MODULE)
++find_package(LibXml2 QUIET MODULE)
```
Comment (veprbl) [inline pkgs/development/libraries/libiio/cmake-fix-libxml2-find-package.patch:9]: Would that not be an issue with libxml2 itself?

I do see the config present at /nix/store/7xd3qxnfqd9gy2wxd5i4q2v4r9201imm-libxml2-2.9.12-dev/lib/cmake/libxml2/libxml2-config.cmake

Comment (veprbl) [reply pkgs/development/libraries/libiio/cmake-fix-libxml2-find-package.patch:9]: Seems like a problem with multiple-outputs of libxml2. This one configures fine for me:

```
with import <nixpkgs> {};
  
libiio.override {
  libxml2 = libxml2.overrideAttrs (old: {
    postFixup = old.postFixup + ''
      substituteInPlace "$dev"/lib/cmake/libxml2/libxml2-config.cmake \
        --replace "set(LIBXML2_LIBRARY_DIR " "set(LIBXML2_LIBRARY_DIR \"$out/lib\")#"
    '';
  });
}
```

**Comment by veprbl** [comment]
> Thanks for the fix! Can you please base it against `staging`?

Since libiio is broken on master (https://hydra.nixos.org/eval/1674190?filter=libiio&compare=1674155&full=), this can go to staging-next.

**Comment by veprbl** [comment]
@SCOTT-HAMILTON I'm not sure: are you forbidding maintainer pushes to your PR? Or the rebase action has been broken lately, as this is not the first time I see this. Could you open a PR against master, please?

**Comment by roberth** [comment]
This PR needs a rebase.

It appears that github removed more requests for review; I only intended to remove myself because the list was excessive.

**Pull request closed without merging**